server/src/main/java/org/elasticsearch/action/support/TransportAction.java

```diff
@@ -18,24 +18,26 @@
  */
 
 package org.elasticsearch.action.support;
-
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.message.ParameterizedMessage;
 import org.apache.logging.log4j.Logger;
 import org.elasticsearch.action.ActionListener;
 import org.elasticsearch.action.ActionRequest;
 import org.elasticsearch.action.ActionRequestValidationException;
 import org.elasticsearch.action.ActionResponse;
-import org.elasticsearch.common.component.AbstractComponent;
 import org.elasticsearch.tasks.Task;
 import org.elasticsearch.tasks.TaskListener;
 import org.elasticsearch.tasks.TaskManager;
 
 import java.util.concurrent.atomic.AtomicInteger;
 
-public abstract class TransportAction<Request extends ActionRequest, Response extends ActionResponse> extends AbstractComponent {
+public abstract class TransportAction<Request extends ActionRequest, Response extends ActionResponse> {
 
     protected final String actionName;
     private final ActionFilter[] filters;
     protected final TaskManager taskManager;
+    @Deprecated
+    protected static Logger logger = LogManager.getLogger(TransportAction.class);
```

Comment:
Using "LogManager.getLogger(TransportAction.class)" would still mean all inheriting classes using this would share the same logger. I don't think this is what we want. Something like "LogManager.getLogger(getClass())" might work for now. Also: why did you add the "@Deprecated" annotation? The loggers themselves are not ment to be deprecated as far as I understand #34488 .

Reply:
I like deprecating them because I think everything that wants a logger should declare one. That way we don't have a zillion loggers we don't use. I do think that if you deprecate something you should add javadoc explaining what to do instead. Something like:
```
/**
 * @deprecated declare your own logger.
 */
@Deprecated
protected static Logger logger = LogManager.getLogger(TransportAction.class);
```

Reply:
I will add the comment on deprecated loggers.